go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -532,10 +536,339 @@ func (c *cnxn) GetInfo(ctx context.Context, infoCodes []adbc.InfoCode) (array.Re
+	dbSchemaNameBuilder := catalogDbSchemasItems.FieldBuilder(0).(*array.StringBuilder)
+	dbSchemaTablesBuilder := catalogDbSchemasItems.FieldBuilder(1).(*array.ListBuilder)
+	dbSchemaTablesItems := dbSchemaTablesBuilder.ValueBuilder().(*array.StructBuilder)
+	tableNameBuilder := dbSchemaTablesItems.FieldBuilder(0).(*array.StringBuilder)
+	tableTypeBuilder := dbSchemaTablesItems.FieldBuilder(1).(*array.StringBuilder)
+	tableColumnsBuilder := dbSchemaTablesItems.FieldBuilder(2).(*array.ListBuilder)
+	tableColumnsItems := tableColumnsBuilder.ValueBuilder().(*array.StructBuilder)
+	columnNameBuilder := tableColumnsItems.FieldBuilder(0).(*array.StringBuilder)
+	ordinalPositionBuilder := tableColumnsItems.FieldBuilder(1).(*array.Int32Builder)
+	remarksBuilder := tableColumnsItems.FieldBuilder(2).(*array.StringBuilder)
+	xdbcDataTypeBuilder := tableColumnsItems.FieldBuilder(3).(*array.Int16Builder)
+	xdbcTypeNameBuilder := tableColumnsItems.FieldBuilder(4).(*array.StringBuilder)
+	xdbcColumnSizeBuilder := tableColumnsItems.FieldBuilder(5).(*array.Int32Builder)
+	xdbcDecimalDigitsBuilder := tableColumnsItems.FieldBuilder(6).(*array.Int16Builder)
+	xdbcNumPrecRadixBuilder := tableColumnsItems.FieldBuilder(7).(*array.Int16Builder)
+	xdbcNullableBuilder := tableColumnsItems.FieldBuilder(8).(*array.Int16Builder)
+	xdbcColumnDefBuilder := tableColumnsItems.FieldBuilder(9).(*array.StringBuilder)
+	xdbcSqlDataTypeBuilder := tableColumnsItems.FieldBuilder(10).(*array.Int16Builder)
+	xdbcDatetimeSubBuilder := tableColumnsItems.FieldBuilder(11).(*array.Int16Builder)
+	xdbcCharOctetLengthBuilder := tableColumnsItems.FieldBuilder(12).(*array.Int32Builder)
+	xdbcIsNullableBuilder := tableColumnsItems.FieldBuilder(13).(*array.StringBuilder)
+	xdbcScopeCatalogBuilder := tableColumnsItems.FieldBuilder(14).(*array.StringBuilder)
+	xdbcScopeSchemaBuilder := tableColumnsItems.FieldBuilder(15).(*array.StringBuilder)
+	xdbcScopeTableBuilder := tableColumnsItems.FieldBuilder(16).(*array.StringBuilder)
+	xdbcIsAutoincrementBuilder := tableColumnsItems.FieldBuilder(17).(*array.BooleanBuilder)
+	xdbcIsGeneratedcolumnBuilder := tableColumnsItems.FieldBuilder(18).(*array.BooleanBuilder)
+	tableConstraintsBuilder := dbSchemaTablesItems.FieldBuilder(3).(*array.ListBuilder)
+
+	catalogPattern, err := patternToRegexp(catalog)
+	if err != nil {
+		return nil, adbc.Error{
+			Msg:  err.Error(),
+			Code: adbc.StatusInvalidArgument,
+		}
+	}
+	columnNamePattern, err := patternToRegexp(columnName)
+	if err != nil {
+		return nil, adbc.Error{
+			Msg:  err.Error(),
+			Code: adbc.StatusInvalidArgument,
+		}
+	}
+
+	for rdr.Next() {
+		arr := rdr.Record().Column(0).(*array.String)
+		for i := 0; i < arr.Len(); i++ {
+			catalogName := arr.Value(i)
+			if catalogPattern != nil && !catalogPattern.MatchString(catalogName) {
+				continue
+			}
+			catalogNameBuilder.Append(catalogName)
+
+			if depth != adbc.ObjectDepthCatalogs {
```

Comment:
Can we reduce the levels of indentation by having this be 
```go
if depth == adbc.ObjectDepthCatalogs {
    // do a thing
    continue
}
```

same for the other situations. make the code more readable by doing early `continue`s to reduce the level of indentation.